src/components/core/base/GeoValidatorHOC.tsx

```diff
@@ -43,14 +43,19 @@ export function geoValidatorHOC<T>(InnerComponent: React.ComponentClass<T>): Rea
         }
 
         public shouldComponentUpdate(nextProps: IGeoValidatorProps): boolean {
-            const { config, dataSource } = this.props;
-            const { config: nextConfig, dataSource: nextDataSource } = nextProps;
+            const { config, dataSource, drillableItems } = this.props;
+            const {
+                config: nextConfig,
+                dataSource: nextDataSource,
+                drillableItems: nextDrillableItems,
+            } = nextProps;
 
             // check if buckets, filters and config are changed
             const isSameConfig = this.isSameConfig(config, nextConfig);
             const isSameDatasource = isEqual(dataSource.getAfm(), nextDataSource.getAfm());
+            const isSameDrillableItems = isEqual(drillableItems, nextDrillableItems);
```

Comment:
Just small note.
Should we only compare items which are not predicate?
I mean predicates are arrow functions which `isEqual` always return false, doesn' it?

Reply:
drillableItems is cached here
https://github.com/gooddata/gdc-dashboards/blob/develop/app/modules/Drilling/DrillService.ts#L100
So isEqual check is still valid